src/commands/deploy/java/tryGetMavenModule.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as parser from 'fast-xml-parser';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { IDeployContext } from 'vscode-azureappservice';
+
+type MavenModule = { path: string, artifactId: string, artifactFinalName: string };
+type MavenPom = {
+    project: {
+        parent?: { version?: string },
+        artifactId: string,
+        version?: string,
+        packaging?: string,
+        build?: { finalName?: string }
+    }
+}
+
+export async function tryGetMavenModule(context: IDeployContext, projectPath: string): Promise<MavenModule | undefined> {
+    const pomFile = path.join(projectPath, 'pom.xml');
+    if ((await fse.pathExists(projectPath)) && (await fse.lstat(projectPath)).isDirectory() && (await fse.pathExists(pomFile))) {
+        return getMavenModuleFromPom(pomFile);
+    }
+    return undefined;
+}
+
+async function getMavenModuleFromPom(pomFile: string): Promise<MavenModule | undefined> {
+    const pomContent = await fse.readFile(pomFile, 'utf8');
```

Comment:
Do we need to specify encoding? If you leave out `utf8` I think `readFile` will figure it out for us.

Reply:
the type of returned result is `Promise<String>` if `encoding` is specified, and `Promise<Buffer>` otherwise. and I want `string` here. 
```
// fs-extra/index.d.ts

export function readFile(file: string | Buffer | number, encoding: string): Promise<string>;
export function readFile(file: string | Buffer | number): Promise<Buffer>;
```
and it's same as `fs.readFile`
https://nodejs.org/api/fs.html#fs_fspromises_readfile_path_options
![image](https://user-images.githubusercontent.com/69189193/126887324-9594b1fc-3947-4425-95a1-fc57449d3ae4.png)

Reply:
Yeah we normally just do `.toString()` on the buffer

Reply:
that also makes sense, I added another commit to use `Buffer.toString()` instead.